### PR TITLE
Using dynamic methods to fetch zones.

### PR DESCRIPTION
**Pull request by bathina2**
## Change Overview

The following PR using dynamic methods to fetch zones for GCP, AWS and Azure

## Pull request type

Please check the type of change your PR introduces:
- [x] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues

- #XXX

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/blockstorage/awsebs/awsebs.go [hunk 1]
```diff
@@ -82,7 +83,7 @@ func newEC2Client(awsRegion string, config *aws.Config) (*EC2, error) {
 	}
 	s, err := session.NewSession(config)
 	if err != nil {
-		return nil, errors.Wrap(err, "Failed to create session for EFS")
+		return nil, errors.Wrap(err, "Failed to create session for EBS")
```
Comment (carlbraganza) [inline pkg/blockstorage/awsebs/awsebs.go:86]: Why "EBS"? Why not "Failed to create EC2 session"?

File: pkg/blockstorage/awsebs/awsebs.go [hunk 2]
```diff
@@ -618,8 +619,40 @@ func GetRegionFromEC2Metadata() (string, error) {
 
 // FromRegion is part of zone.Mapper
 func (s *EbsStorage) FromRegion(ctx context.Context, region string) ([]string, error) {
-	// Fall back to using a static map.
-	return staticRegionToZones(region)
+	ec2Cli, err := newEC2Client(region, s.config)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Could not get EC2 client")
+	}
```
Comment (carlbraganza) [inline pkg/blockstorage/awsebs/awsebs.go:625]: This wrap adds nothing of value. The message should identify the context of the failure (in this case the creation of the client in FromRegion).

File: pkg/blockstorage/awsebs/awsebs.go (hunk 1 above)
Comment (bathina2) [reply pkg/blockstorage/awsebs/awsebs.go:86]: I just fixed the some copy paste error in the past. It definitely shouldn't say EFS

Comment (bathina2) [reply pkg/blockstorage/awsebs/awsebs.go:86]: This is getting the Ec2client for the EBS provider.

**Review by onkarbhat — APPROVED** [review]
LGTM :shipit:  🚢

**Pull request merged**